src/app.js

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2015 Colin Clark
+
+Licensed under the 3-Clause "New" BSD license.
+You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the 3-Clause "New" BSD License at
+https://github.com/colinbdclark/infusion-electron/raw/master/LICENSE.txt
+*/
+
+"use strict";
+
+var fluid = require("infusion"),
+    electron = fluid.registerNamespace("electron");
+
+// Monkey-patch fluid.log for Electron's broken console.log.
+fluid.doLog = function (args) {
+    if (typeof (console) !== "undefined") {
+        if (console.debug) {
+            console.debug(args.join(" "));
+        } else if (typeof (console.log) === "function") {
+            console.log(args.join(" "));
+        }
+    }
+};
+
+
+fluid.defaults("electron.app", {
+    gradeNames: "fluid.modelComponent",
+
+    members: {
+        app: require("app")
+    },
+
+    commandLineSwitches: {},
+
+    windowListeners: {
+        "onClose": "electron.app.quitOnAllClosed()"
+    },
+
+    events: {
+        onReady: null,
+        onAllWindowsClosed: null
+    },
+
+    listeners: {
+        onCreate: [
```

Comment:
These are much better namespaced so that the component can be more accurately configured by others - "onCreate.setCommandLineSwitches" and "onCreate.bindAppEvents" - the latter two entries are better hived off into a function of their own